test/widgets/home_test.dart

```diff
@@ -0,0 +1,409 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_checks/flutter_checks.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/api/model/events.dart';
+import 'package:zulip/model/store.dart';
+import 'package:zulip/widgets/app.dart';
+import 'package:zulip/widgets/app_bar.dart';
+import 'package:zulip/widgets/home.dart';
+import 'package:zulip/widgets/icons.dart';
+import 'package:zulip/widgets/inbox.dart';
+import 'package:zulip/widgets/page.dart';
+import 'package:zulip/widgets/profile.dart';
+import 'package:zulip/widgets/subscription_list.dart';
+import 'package:zulip/widgets/theme.dart';
+
+import '../api/fake_api.dart';
+import '../example_data.dart' as eg;
+import '../flutter_checks.dart';
+import '../model/binding.dart';
+import '../model/test_store.dart';
+import 'page_checks.dart';
+import 'test_app.dart';
+
+void main () {
+  TestZulipBinding.ensureInitialized();
+
+  late PerAccountStore store;
+
+  Future<void> prepare(WidgetTester tester) async {
+    addTearDown(testBinding.reset);
+    await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+    store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+
+    await store.addUsers([eg.selfUser, eg.otherUser]);
+    final stream = eg.stream();
+    await store.addStream(stream);
+    await store.addSubscription(eg.subscription(stream));
+
+    await store.handleEvent(MessageEvent(
+      id: 0, message: eg.streamMessage(stream: stream, sender: eg.otherUser)));
```

Comment:
This feels awfully specific. Is it generally needed?

Looks like just one test fails if I delete it. Can that test do this for itself?

Reply:
It's for preparing a message in the inbox. We can either access `PerAccountStore` from that test, keep the `store.handleEvent` inside, or find a different state to test.

Reply:
Yeah, accessing the store from that test sounds good.

Reply:
(and then I guess it'll stay a late, shared variable)